test/utils.py

```diff
@@ -79,37 +79,58 @@ def wait_for_event(self, event, count):
         wait_until(lambda: self.event_count(event) >= count, "find %s %s event(s)" % (count, event))
 
 
+def log_func_call(func):
+    def inner(self, *args):
+        self.func_calls.append(func.__name__)
+        func(self, *args)
+
+    return inner
+
+
 class CMAPListener(BaseListener, monitoring.ConnectionPoolListener):
+    func_calls = []
+
+    @log_func_call
```

Comment:
What do you think about asserting the proper arg type in the method itself (rather than after the fact):
```python
    def connection_created(self, event):
        assert isinstance(event, ConnectionCreatedEvent)
        self.add_event(event)
```

Note that the tests might be failing due to a difference in behavior of load balanced vs non-load balanced pools. Or it could be another bug.